apps/req.c

```diff
@@ -558,24 +558,20 @@ int req_main(int argc, char **argv)
         X509V3_CTX ctx;
 
         X509V3_set_ctx_test(&ctx);
-        X509V3_set_nconf(&ctx, req_conf);
-        if (!X509V3_EXT_add_nconf(req_conf, &ctx, extsect, NULL)) {
-            BIO_printf(bio_err,
-                       "Error checking %s extension section %s\n",
-                       gen_x509 ? "x509" : "request", extsect);
+        if (!do_EXT_add_nconf(req_conf, &ctx, extsect, NULL,
+                              !gen_x509
+                              ? "Error checking request extension section %s\n"
+                              : "Error checking x509 extension section %s\n"))
             goto end;
-        }
     }
     if (addext_conf != NULL) {
         /* Check syntax of command line extensions */
         X509V3_CTX ctx;
 
         X509V3_set_ctx_test(&ctx);
-        X509V3_set_nconf(&ctx, addext_conf);
-        if (!X509V3_EXT_add_nconf(addext_conf, &ctx, "default", NULL)) {
-            BIO_printf(bio_err, "Error checking extensions defined using -addext\n");
+        if (!do_EXT_add_nconf(addext_conf, &ctx, "default", NULL,
+                              "Error checking x509 extensions defined via -addext%.0s\n"))
```

Comment:
This is really ugly. Could you just make the printf in do_EXT_add_nconf() to not add the sect if it is NULL? But then it risks problems if sect value can be NULL in other cases... Perhaps add another parameter?

Reply:
All right, changed this to allow a NULL `sect` arg.
I think for such an app-internal function it is bearable that the `msg` arg must contain `%s` if and only if the `sect` arg is non-NULL.